controllers/release/release_adapter.go

```diff
@@ -213,54 +214,59 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.Operation
 		return results.ContinueProcessing()
 	}
 
-	err = a.syncResources()
+	environment, err := a.getEnvironment(releasePlanAdmission)
 	if err != nil {
 		return results.RequeueWithError(err)
 	}
 
-	binding, err := a.createOrUpdateSnapshotEnvironmentBinding(releasePlanAdmission)
+	err = a.syncResources()
 	if err != nil {
 		return results.RequeueWithError(err)
 	}
 
-	a.logger.Info("Created/updated SnapshotEnvironmentBinding",
-		"SnapshotEnvironmentBinding.Name", binding.Name, "SnapshotEnvironmentBinding.Namespace", binding.Namespace)
+	// Search for an existing binding
+	binding, err := a.getSnapshotEnvironmentBinding(environment, releasePlanAdmission)
+	if err != nil && !errors.IsNotFound(err) {
+		return results.RequeueWithError(err)
+	}
 
 	patch := client.MergeFrom(a.release.DeepCopy())
-	a.release.Status.SnapshotEnvironmentBinding = fmt.Sprintf("%s%c%s", binding.Namespace, types.Separator, binding.Name)
+
+	if binding == nil {
+		binding, err := a.createSnapshotEnvironmentBinding(releasePlanAdmission, environment)
+		if err != nil {
+			return results.RequeueWithError(err)
+		}
+
+		a.logger.Info("Created/updated SnapshotEnvironmentBinding",
+			"SnapshotEnvironmentBinding.Name", binding.Name, "SnapshotEnvironmentBinding.Namespace", binding.Namespace)
+
+		a.release.Status.SnapshotEnvironmentBinding = fmt.Sprintf("%s%c%s", binding.Namespace, types.Separator, binding.Name)
+	}
 
 	return results.RequeueOnErrorOrContinue(a.client.Status().Patch(a.context, a.release, patch))
```

Comment:
The release is only modified inside the `if` so the patch and the update of the status should go inside as well.